cargo-dylint/tests/ci.rs

```diff
@@ -713,6 +715,103 @@ fn update() {
 }
 
+#[test]
+fn lint_script_test() {
+    preserves_cleanliness("lint_script_test", false, || {
+        let cargo_dylint_path = Path::new("target/debug/cargo-dylint")
+            .canonicalize()
+            .expect("Failed to find cargo-dylint executable. Ensure it is built, e.g., with `cargo build -p cargo-dylint`.");
+
+        let example_restriction_dir = Path::new("examples/restriction");
+        let mut restriction_libs = Vec::new();
+        if example_restriction_dir.is_dir() {
+            for entry in read_dir(example_restriction_dir).unwrap() {
+                let entry = entry.unwrap();
+                if entry.path().is_dir() {
+                    let lib_name = entry.file_name().into_string().unwrap();
+                    // Exclude overscoped_allow as in the script
+                    if lib_name != "overscoped_allow" && lib_name != ".cargo" {
+                        restriction_libs.push(format!("--lib {lib_name}"));
+                    }
+                }
+            }
+        }
+        let restrictions_as_flags = restriction_libs.join(" ");
+
+        let mut experimental_dirs_str = Vec::new();
+        let example_experimental_dir = Path::new("examples/experimental");
+        if example_experimental_dir.is_dir() {
+            for entry in read_dir(example_experimental_dir).unwrap() {
+                let entry = entry.unwrap();
+                let path = entry.path();
+                if path.is_dir() && path.file_name().unwrap() != ".cargo" {
+                    // Convert path to string for the DIRS array
+                    experimental_dirs_str.push(path.to_string_lossy().to_string());
+                }
+            }
+        }
+
+        let base_flags =
+            format!("--lib general --lib supplementary {restrictions_as_flags} --lib clippy");
+
+        let mut dirs_to_lint = vec![
+            ".",
+            "driver",
+            "utils/linting",
+            "examples/general",
+            "examples/supplementary",
+            "examples/restriction",
+            "examples/testing/clippy",
+        ]
+        .into_iter()
+        .map(String::from)
+        .collect::<Vec<_>>();
+
+        dirs_to_lint.extend(experimental_dirs_str);
+
+        // Function similar to force_check in lint.sh
+        let workspace_root = Path::new("."); // current dir is workspace root
+        let target_dir = workspace_root.join("target");
+        if target_dir.exists() {
+            for entry in walkdir::WalkDir::new(&target_dir) {
+                let entry = entry.unwrap();
+                if entry.file_name().to_string_lossy() == ".fingerprint"
+                    && (entry
+                        .path()
+                        .starts_with(target_dir.join("dylint/target/nightly-"))
+                        || entry.path().starts_with(target_dir.join("nightly-")))
+                {
+                    let _ = remove_dir_all(entry.path());
+                }
+            }
+        }
```

Comment:
```suggestion
```
I know you did this because lint.sh does this. But that lint.sh code was added to support `overscoped_allow`, which is no longer run. So let's please try this test without that code.